dds/DCPS/GuidUtils.cpp

```diff
@@ -30,17 +30,54 @@ setopts(std::ostream& os)
 {
   return os << std::setfill('0') << std::setw(2);
 }
+#endif
 
 } // namespace
 
 namespace OpenDDS { namespace DCPS {
 
+OPENDDS_STRING
+to_string(const GUID_t& guid)
+{
+  std::size_t len;
+  std::size_t tot_len;
+
+  tot_len = sizeof(guid.guidPrefix) +
```

Comment:
I don't see how the sizeof() these various things directly gets us the string length.  Pre-computing the string length is not strictly necessary.  If it's more trouble than it's worth, just let the string grow incrementally.